### PR TITLE
Add profile views counter

### PR DESCRIPTION
**Pull request by Palakkgoyal**
This PR closes issue no. #543 
Implemented a profile view counter and it does not update on every render rather I have used session storage to check if the views are updated and if not then, update the views only once in a particular session.

### CODE REVIEW

File: src/reusableComponents/ViewsCounter.jsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            sessionStorage.getItem("viewsCount"),
+            10
+          );
+          setViews(currentViews);
+          return;
+        }
+
+        const querySnapshot = await db
+          .collection("profileViews")
+          .where("uid", "==", auth.currentUser.uid)
```
Comment (narayan954) [inline src/reusableComponents/ViewsCounter.jsx:23]: won't this just update the user's view of his own profile only?

**Review by narayan954 — CHANGES_REQUESTED** [review]
check the comment, will look into this again today

**Comment by Palakkgoyal** [comment]
Let me go through it once again.

**Comment by Palakkgoyal** [comment]
Yes, you are right. I think we should use username for all these checkings.

**Comment by narayan954** [comment]
> Yes, you are right. I think we should use username for all these checkings.

yes, it seems it's not that simple.

**Comment by Palakkgoyal** [comment]
Yes, I initially thought it would be easy to implement. But, no worries, I will give my best.

**Comment by narayan954** [comment]
> Yes, I initially thought it would be easy to implement. But, no worries, I will give my best.

Best of luck! :)

**Comment by narayan954** [comment]
is this done?

**Comment by Palakkgoyal** [comment]
Not yet, I thought it would be better if I use username instead of display name. So, for sometime I stopped working on this.

**Comment by narayan954** [comment]
> Not yet, I thought it would be better if I use username instead of display name. So, for sometime I stopped working on this.

Ohh alright! you did right!

**Pull request closed without merging**